Lib/test/test_zipimport.py

```diff
@@ -933,6 +933,20 @@ def assertDataEntry(name):
 
         self.doTestWithPreBuiltZip(".py", "module")
 
+    def testImportSubmodulesInZip(self):
+        with ZipFile(TEMP_ZIP, "w") as z:
+            z.mkdir("a")
+            z.writestr("a/__init__.py", b'')
+            z.mkdir("a/b")
+            z.mkdir("a/b/c")
+            z.writestr("a/b/c/__init__.py", b'def foo(): return "foo"')
+
+        importer = zipimport.zipimporter(TEMP_ZIP)
+        spec = importer.find_spec("a.b.c")
+        mod = importlib.util.module_from_spec(spec)
+        importer.exec_module(mod)
+        self.assertEqual(mod.foo(), "foo")
+
```

Comment:
If here test the namespace module in zip, `mkdir()` must be removed, this bug is triggered by no directory entry in zip.

And don't do test via zipimport, that always fail/succeed, because find functons depend the loader's path prefix. Direct import namespace module can cover all cases.

Reply:
I'm a bit confused on what you want here - the test passes, why should `mkdir()` be removed?

Reply:
Again, this bug is triggered by no directory entry in zip. If the directory entry is already exists, everything is OK.